src/components/PopupWithForm.js

```diff
@@ -14,13 +15,19 @@ export default class PopupWithForm extends Popup {
     return inputValues;
   }
 
+  open() {
+    this._submitButton.textContent = "Save";
+    super.open();
+  }
+
   close() {
     this._popupForm.removeEventListener("submit", this._handleSubmitButton);
     this._popupForm.reset();
     super.close();
   }
 
   _handleSubmitButton = (e) => {
+    this._submitButton.textContent = "Saving...";
```

Comment:
Needs correction: please don't change the button text here. You should create a separate method for rendering loading and start the loading right before the api request and end the loading after in `finally` block